markets/perps-market/test/integration/Orders/OffchainAsyncOrder.commit.test.ts

```diff
@@ -86,6 +87,28 @@ describe('Commit Offchain Async Order test', () => {
         'InsufficientMargin'
       );
     });
+
+    it('reverts if max market value is reached', async () => {
+      await depositCollateral({
+        accountId: () => 2,
+        collaterals: [{ snxUSDAmount: () => bn(100_000) }],
+        systems,
+        trader: trader1,
+      });
+      await assertRevert(
+        systems()
+          .PerpsMarket.connect(trader1())
+          .commitOrder({
+            marketId: ethMarketId,
+            accountId: 2,
+            sizeDelta: bn(90_000),
+            settlementStrategyId: 0,
+            acceptablePrice: bn(1450), // 5% slippage
+            trackingCode: ethers.constants.HashZero,
+          }),
+        `MaxOpenInterestReached(${ethMarketId}, ${bn(5000).toString()})`
+      );
+    });
```

Comment:
unfortunately, there are many other cases to be tested here. for one, we don't test if there's a negative sizes.